honeybee_vtk/to_vtk.py

```diff
@@ -100,32 +100,87 @@ def convert_sensor_grid(
     return grid_data
 
 
+def _add_metadata_to_cells(polydata, hb_object, hb_type: str = None) -> PolyData:
+    """This is an internal method to add metadata from the Polydata to its cells.
+
+    Args:
+        polydata: A honeybee-vtk Polydata object.
+        hb_object: A honeybee object. A Face, Aperture or Shade.
+        hb_type: A string representing the face type. Acceptable values are 'Aperture',
+            'Shade'.
+
+    Returns:
+        A honeybee-vtk Polydata object with the metadata.
+    """
+    # setting polydata properties
+    if not hb_type:
+        polydata.type = hb_object.type.name
+    else:
+        polydata.type = hb_type
+    polydata.identifier = hb_object.identifier
+    polydata.display_name = hb_object.display_name
+    if hb_type != 'Shade':
+        polydata.boundary_condition = hb_object.boundary_condition.ToString()
+    polydata.construction_display_name = hb_object.properties.energy.construction.display_name
+    polydata.modifier_display_name = hb_object.properties.radiance.modifier.display_name
+
+    # number of cells in the polydata
+    num_of_cells = polydata.GetNumberOfCells()
+
+    # Applying string metdata to the cells
+    # is hb objects is a shade avoid boundary condition
+    if hb_type == 'Shade':
+        display_name = [polydata.display_name] * num_of_cells
+        construction_display_name = [polydata.construction_display_name] * num_of_cells
+        modifier_display_name = [polydata.modifier_display_name] * num_of_cells
+
+        metadata = {
+            "Display Name": display_name,
+            "Construction Display Name": construction_display_name,
+            "Modifier Display Name": modifier_display_name
+        }
+    else:
+        display_name = [polydata.display_name] * num_of_cells
+        boundary_condition = [polydata.boundary_condition] * num_of_cells
+        construction_display_name = [polydata.construction_display_name] * num_of_cells
+        modifier_display_name = [polydata.modifier_display_name] * num_of_cells
+
+        metadata = {
+            "Display Name": display_name,
+            "Boundary Condition": boundary_condition,
+            "Construction Display Name": construction_display_name,
+            "Modifier Display Name": modifier_display_name
```

Comment:
As I commented on Basecamp I think we can simplify the keys and use name, boundary condition, construction and modifier.

Reply:
Complied